core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -819,12 +819,14 @@ wasmtime_ssp_fd_prestat_dir_name(struct fd_prestats *prestats, __wasi_fd_t fd,
         rwlock_unlock(&prestats->lock);
         return error;
     }
-    if (path_len != strlen(prestat->dir)) {
+
+    const size_t prestat_dir_len = (size_t)strlen(prestat->dir);
```

Comment:
Had better use `strlen(prestat->dir) + 1` to contain `\0` for he below path_len check and bh_memcpy_s?
And no need to add `(size_t)`, the return type of strlen is already size_t.

Reply:
iirc, the name here isn't NUL-terminated.

Reply:
Yes, if it isn't '\0' terminated, we can merge the PR.
@zoraaver Could you help confirm it?

Reply:
Hi @wenyongh, I can confirm that the input `path` is not null-terminated. `path_len` should be obtained from [`fd_prestat_get`](https://docs.rs/wasi/latest/wasi/struct.PrestatDir.html) which doesn't not include the null character [in our case]( https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c#L804).

I've removed the cast, thanks for spotting!